### PR TITLE
Minor changes to linear regression (mostly code cleanup) and util function now generates vector in range 0-1024

### PR DESCRIPTION
**Pull request by fasiddique**

### CODE REVIEW

File: PIMbench/util.h [hunk 1]
```diff
@@ -21,14 +21,16 @@
...
+
 void getVector(uint64_t vectorLength, std::vector<int> &srcVector)
 {
   srand((unsigned)time(NULL));
   srcVector.resize(vectorLength);
 #pragma omp parallel for
   for (uint64_t i = 0; i < vectorLength; ++i)
   {
-    srcVector[i] = (rand() % (i + 1) + 1);
+    srcVector[i] = (rand() % MAX_NUMBER);
```
Comment (KumareshN93) [inline PIMbench/util.h:33]: Do we also need to update getMatrix() function  as above..

File: PIMbench/cpp-linear-regression/lr.cpp [hunk 2]
```diff
@@ -203,34 +174,34 @@ int main(int argc, char *argv[])
...
+  auto intercept_device = (double)(SY_device - slope_device * SX_device) / params.dataSize;
   auto end = std::chrono::high_resolution_clock::now();
   hostElapsedTime += (end - start);
 
   if (params.shouldVerify)
   {
     // verify result
-    int SX = 0, SY = 0, SXX = 0, SYY = 0, SXY = 0;
+    int64_t SX = 0, SY = 0, SXX = 0, SXY = 0;
 #pragma omp parallel for reduction(+ : SX, SXX, SY, SYY, SXY)
```
Comment (KumareshN93) [inline PIMbench/cpp-linear-regression/lr.cpp:192]: Should we remove SYY from the pragma?

File: PIMbench/cpp-linear-regression/lr.cpp [hunk 3]
```diff
@@ -203,34 +174,34 @@ int main(int argc, char *argv[])
...
       SXY += dataPointsX[i] * dataPointsY[i];
     }
+
     // Calculate slope and intercept
-    auto slope = (params.dataSize * SXY - SX * SY) / (params.dataSize * SXX - SX * SX);
-    auto intercept = (SY - slope * SX) / params.dataSize;
-    if (intercept != intercept_device)
+    auto slope = (double)(params.dataSize * SXY - SX * SY) / (params.dataSize * SXX - SX * SX);
+    auto intercept = (double)(SY - slope * SX) / params.dataSize;
+    if ((int)intercept != intercept_device)
```
Comment (KumareshN93) [inline PIMbench/cpp-linear-regression/lr.cpp:204]: Quick question. intercept is typecast to int while intercept_device is double. Is this expected?

**Review by KumareshN93 — APPROVED** [review]
Changes look good.

File: PIMbench/util.h (hunk 1 above)
Comment (fasiddique) [reply PIMbench/util.h:33]: Updated thanks!

**Pull request merged**